sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDataWritingCommandExec.scala

```diff
@@ -54,6 +58,37 @@ trait GpuDataWritingCommand extends DataWritingCommand with ShimUnaryCommand {
   def requireSingleBatch: Boolean
 }
 
+object GpuDataWritingCommand {
+  private val allowNonEmptyLocationInCTASKey = "spark.sql.legacy.allowNonEmptyLocationInCTAS"
+
+  private def getAllowNonEmptyLocationInCTAS: Boolean = {
+    // config only exists in Spark 3.2+, so looking it up manually for now.
```

Comment:
So I am a bit conflicted here because we are not consistent with how we handle new configs. Could we please come up with some rules/guidelines that we can use to decide when to add a config to the shim layer, and when to just hard code the config.

The pros for doing it this way is that there is no shim layer changes needed. The cons are that if for some reason the user has the config set, possibly because they are switching back and forth between versions of Spark, the behavior is not going to be consistent with the version of Spark.

For me I see both sides of it and I don't care which we use, but I want us to be consistent in how we use do it.

Reply:
The reason I didn't shim this one is because it's related to data loss and there's a workaround emitted in the error message if it comes up in practice.  I'm a bit surprised the Spark fix wasn't backported to earlier versions as as result.  I'm happy to make this a shim thing if we feel that's the better approach.

Reply:
My goal is mainly that we have a consistent set of rules. I am fine with us doing different things in different situations, but I do want it to be documented and consistent. Can we have a follow on issue to look at figuring out these rules?

Reply:
Filed #3427